### PR TITLE
docs: documentation refresh for milestone

### LINKED ISSUE #566: Update documentation for beta-4 milestone
We should ensure that our documentation is up to date with information about recently added features associated with our beta-4 milestone. Said information doesn't have to be completely fleshed out, but we should make it a point to include at least something about each part.

Running list of things of undocumented features that are implemented or waiting for review:
- [ ] Native execution
- [x] Optional schema fields
- [x] `forc postgres` (added as part of #558)
- [x] ~Nested API queries (added as part of #563)~
- [ ] New Sway types (?)
- [x] Option to start local node with indexer binary (added as part of #551)
- [x] Add information about changing the contract ID in a manifest after editing the Sway code

### PR DESCRIPTION
**Pull request by deekerno**
Closes #566, #579.

## Changelog
- Add information about contract IDs and the need to change them after editing contract
- Add information about optional schema fields
- Remove bits using `--background` flag with `fuel-indexer run` command
- Fix query example on block explorer example
- Add link to M1/M2 WASM fix to WASM dependency page
- Fix keyboard navigation issue (#579) by adding index pages to problem sections
- Fix terminology usage in places
- Add "Queries" page to GraphQL (I felt that it's better to have more information than less)
- Add Appendix header with Glossary section
- Edit and correct sentence structure in various places
- Replaced old Fuel logo in `README` with a Fuel Indexer header image (thanks @Braqzen!)

## Testing Steps
0. `cargo install mdbook`
1. `cd docs`
2. `mdbook serve`
3. Navigate to `localhost:3000` and read the book

### CODE REVIEW

**Review by sarahschwartz — COMMENTED** [review]
Amazing, thank you! For the receipt type references it would be nice to have small examples for how those can be used in a handler function. For example, in the ReturnData receipt type, we could add something like this:

You can handle functions that produce a `ReturnData` receipt type by using the returned type as a function parameter like this:

```rust, ignore
fn handle_return_data(data: MyStruct) {
  // handle the returned data
}
```

**Comment by deekerno** [comment]
@sarahschwartz, I definitely think that's a beneficial change! If @ra0x3 agrees, then feel free to push up something that you feel works best.

File: docs/src/reference-guide/components/graphql/directives.md [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-
-Using our `Library` and `Book` example from the previous [Foreign Keys](../database/foreign-keys.md) section -- given the following schema:
+- `@indexed`
+- `@unique`
+- `@join`
 
 ## `@indexed`
 
-The `@indexed` directive adds an index to the underlying database column for the indicated field of that type. Generally, an index is a data structure that allows you to quickly locate data without having to search each row in a database table.
+The `@indexed` directive adds a database index to the underlying column for the indicated field of that type. Generally, a database index is a data structure that allows you to quickly locate data without having to search each row in a database table.
```
Comment (rvmelkonian) [inline docs/src/reference-guide/components/graphql/directives.md:14]: this is referring to postgres indexes, I think it's worth explicitly noting that this has nothing to do with a fuel index, or link to this [https://www.postgresql.org/docs/current/indexes.html](), so a user could see exactly what this is and quickly understand the difference.

File: docs/src/reference-guide/components/indexers/index.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Indexers
+
+Indices are the one of the core building blocks of the Fuel indexer service. Indices allow indexer operators and dApp authors to index specific blockchain events into a database.
```
Comment (rvmelkonian) [inline docs/src/reference-guide/components/indexers/index.md:3]: In `docs/src/the-fuel-indexer.md` we refer to Indexer plural as Indexers, here we say Indices. Let's stay consistent, I don't mind which one we choose, personally I prefer Indexers. Also, The title of this section is # Indexers then we immediately say `Indices`, a bit confusing.

**Review by rvmelkonian — COMMENTED** [review]
Some really great additions to our docs! Particularly `graphql/queries.md`, I think users will find that very useful. Just a few suggestions.

File: docs/src/reference-guide/components/indexers/index.md (hunk 2 above)
Comment (deekerno) [reply docs/src/reference-guide/components/indexers/index.md:3]: Oh whoops, that was a find-and-replace gone wrong. Will fix.

File: docs/src/reference-guide/components/assets/manifest.md [hunk 3]
```diff
@@ -16,39 +16,38 @@ report_metrics: true
...
+The `namespace` is the topmost organizational level of an indexer. You can think of different `namespace`s as separate and distinct collections comprised of indexers. The `namespace` is unique to a given indexer operator -- i.e., indexer operators will not be able to support more than one `namespace` of the same name.
 
 ## `identifier`
 
-- The `identifier` field is used to (quite literally) identify the given index. If `namespace` is the organization/domain name, then think of `identifier` as the name of an index within that organization/domain.
-- As an example, if a provided `namespace` is `"fuel"` and a provided `identifier` is `"index1"`, then the unique identifier for the given index will be `fuel.index1`.
+The `identifier` field is used to (quite literally) identify the given indexer. If a `namespace` describes a collection of indexers, then an `identifier` describes a unique indexer inside that collection. As an example, if a provided `namespace` is `"fuel"` and a provided `identifier` is `"index1"`, then the full unique identifier for the given indexer will be `fuel.index1`.
```
[the remark this answers is not in the document]
Comment (deekerno) [reply docs/src/reference-guide/components/assets/manifest.md:23]: I'll remove "unique" here. 👍🏽

File: docs/src/reference-guide/components/assets/manifest.md [hunk 4]
```diff
@@ -16,39 +16,38 @@ report_metrics: true
 
 ## `namespace`
 
-- Think of the `namespace` as an organization identifier. If you're familiar with say, [Java package naming](https://stackoverflow.com/questions/6247849/java-package-naming), then think of an index's `namespace` as being its _domain name_. The `namespace` is unique to a given index operator -- i.e., index operators will not be able to support more than one `namespace` of the same name.
+The `namespace` is the topmost organizational level of an indexer. You can think of different `namespace`s as separate and distinct collections comprised of indexers. The `namespace` is unique to a given indexer operator -- i.e., indexer operators will not be able to support more than one `namespace` of the same name.
```
[the remark this answers is not in the document]
Comment (deekerno) [reply docs/src/reference-guide/components/assets/manifest.md:19]: #1 is good, but I don't think #2 is the correct way; actually, I think we should just do `namespace` at first but then just use "namespace" (without quotes) after that.

File: docs/src/reference-guide/components/assets/index.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # Assets
```
[the remark this answers is not in the document]
Comment (deekerno) [reply docs/src/reference-guide/components/assets/index.md:1]: Nope! The `index.md` file is for `mdbook` to know what should be rendered for the chapter header.

**Comment by ra0x3** [comment]
blocked by https://github.com/FuelLabs/fuel-indexer/pull/630

File: docs/src/reference-guide/components/graphql/queries.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
+# Queries
```
Comment (ra0x3) [inline docs/src/reference-guide/components/graphql/queries.md:1]: @deekerno 
- I think we can remove this. 
- Many (if not all) users expect this to be default behavior. 
- If it turns out we need to explain this we can revive this, but at present, it looks like another piece of documentation that demonstrates functionality most users will expect by default.

File: docs/src/reference-guide/components/graphql/schema.md [hunk 7]
```diff
@@ -30,3 +30,11 @@ The types you see above (e.g., `ID`, `UInt8`, etc) are Fuel abstractions that we
 types can be found in [the Types section](../../data-types/types.md).
 
 > Important: It is up to developers to manage their own unique IDs for each type, meaning that a data structure's `ID` field needs to be manually generated prior to saving it to the database. This generation can be as simple or complex as you want in order to fit your particular situation; the only requirement is that the developer implement their own custom generation. Examples can be found in the [Block Explorer](../../../examples/block-explorer.md) and [Hello World](../../../examples/hello-world.md) sections.
+
+## Required and Optional Fields
```
Comment (ra0x3) [inline docs/src/reference-guide/components/graphql/schema.md:34]: 👌🏽

**Review by ra0x3 — APPROVED** [review]

**Review by sarahschwartz — APPROVED** [review]

**Review by rvmelkonian — APPROVED** [review]

**Pull request merged**